### PR TITLE
Delivering outputs for all task

### PR DESCRIPTION
**Pull request by thebere**
#### What does this PR do?
Complete outputs for
1. Automating task
2. Checking repository health
3. Delivering product
4. Getting peer reviews
5. Reflecting on the learning experience
6. Setting up continuous integration
7. Setting up linters
8. Setting up software deployment environments
9. Setting up repository
10. Solving the problem
11. Using design patterns
12. Writing html, css, and js outputs
13. Writing tests.

#### Description of Task to be completed?
**1. Automating task** A screencast showing gulp as my task runner, restarting my server, with a live reload of the browser when task are made to html, css, and js, files.
**2. Checking repository health** A screenshot showing coveralls test coverage badge and codeclimate GPA on my repository's readme.
**3. Delivering product*** A google document showing product release note
**4. Getting peer reviews:** a Pull Request showing at least two Peer Code Reviews
**5. Reflecting on the learning experience*** A [link](https://medium.com/@ajudensi/reflecting-on-my-learning-experience-completing-checkpoint-i-54f3c3a581bd) to a medium post reflecting on my learning experience
**6. Setting up continuous integration** a link to a successful TravisCI build of your development branch with test runs.
7. **Setting up linters** two screenshots of ESLint Airbnb style violations in the terminal and HoundCI comment on a PR.
**8. Setting up software deployment environments** live URLs to a review app, your staging & production apps.
**9. Setting up repository**  deliver the repository for the project and at least one Pull Request.
**10. Solving the problem** deliver the file containing the implementation including all relevant dependencies. (e.g. InvertedIndex class).
**11 Using design patterns**
**12. Writing html, css, and js outputs** a folder containing the template for the proposed frontend of your project.
**13 Writing tests.**  test files for the Inverted Index class and all associated dependencies.

#### How should this be manually tested?
**1. Automating task** [Automating task 1](https://www.youtube.com/watch?v=pVn080h3Y0o),
 [Automating task 2](https://www.youtube.com/watch?v=8RpK1UMfzE4)
**2. Checking repository health** [Link](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-judensi/06-building-a-program-start-to-finish/checking-repository-health/output)
**3. Delivering product** [Release note](https://docs.google.com/document/d/1FRO-L9GTJ2SQ4wiRNl1l2nPUUEtYjAQG7kkKLLXewVI/edit)
**4. Getting peer reviews:** (Peer reviews)[https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-judensi/06-building-a-program-start-to-finish/getting-peer-reviews/output]
**5. Reflecting on learning experience** [Medium post](https://medium.com/@ajudensi/reflecting-on-my-learning-experience-completing-checkpoint-i-54f3c3a581bd)
**6. Setting up continuous integration** [Travis build](https://travis-ci.org/Andela-JUdensi/Inverted-Index/jobs/216849753)
**7. Setting up linters** [Link](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-judensi/06-building-a-program-start-to-finish/setting-up-linters/output)
**8. Setting up software deployment environments** [Review app](https://idexii-review.herokuapp.com/), [Staging app](http://idexii-staging.herokuapp.com/), [Production app](http://idexii.herokuapp.com/) 
**9. Setting up repository**  [Repository link](https://github.com/Andela-JUdensi/Inverted-Index), with a comprehensive README.md file.
**10. Solving the problem** [InvertedIndex](https://github.com/andela/test-simulations-mgmt-repo/blob/bayo-kakashi-judensi/06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js) with a descriptive JSDOC for each method and property.
**11. Using design patterns** [Link](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-judensi/06-building-a-program-start-to-finish/using-design-patterns/output)
**12. Writing html, css, and js outputs:**  [Link](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-judensi/06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output)
**13 Writing tests**  [InvertedIndex class test](https://github.com/andela/test-simulations-mgmt-repo/blob/bayo-kakashi-judensi/06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js). This test uses Jasmine and Expect.js as the assertion library.

#### Any background context you want to provide?
1. Local test coverage is also provided using istanbul and coveralls

#### What are the relevant pivotal tracker stories?
N/A
##### Hound CI error
screen shot 2017-03-30 at 1 14 53 pm

##### Eslint errors
screen shot 2017-03-30 at 1 32 01 pm

##### Codeclimate and CI badges
screen shot 2017-03-30 at 1 37 40 pm

#####  Successful Travis Build
screen shot 2017-03-30 at 1 58 13 pm

##### IDex - Application interface
screen shot 2017-03-31 at 1 18 45 pm

##### iDex - Application interface with index created
screen shot 2017-03-31 at 2 54 59 pm

##### iDex - Application interface with search result for multiple files
screen shot 2017-03-31 at 2 57 12 pm

#### Questions:
N/A

### CODE REVIEW

File: 06-building-a-program-start-to-finish/automating-tasks/output/README.md
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/automating-tasks/output/README.md:10]: Cool

File: 06-building-a-program-start-to-finish/checking-repository-health/output/README.md
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/checking-repository-health/output/README.md:11]: I see your badges

File: 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md:17]: Did you get clarity on the **_peers_** to get reviews from?

File: 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/outcome/README.md
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/outcome/README.md:8]: Good read. 👍

File: 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md:8]: Nice

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md:9]: Nice. Live URLs

File: 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md:7]: Nice

File: 06-building-a-program-start-to-finish/using-design-patterns/output/README.md
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/using-design-patterns/output/README.md:75]: Cool

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: Is this absolutely necessary?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: You have a missing JSDoc comment for this method.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: Your parameter name should be in camelCase to aid readability.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: I believe there are better ways of handling this. You can have one regular to expression to handle this or chain your multiple calls to `.replace()`

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: Error here - Expected JSDoc for '_bookname_' but found '_book_'.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: Please use camelCase for your variable/parameter names. It aids readability.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: Did you get a multiple instances of this error _**Expected to return a value in arrow function**_ on this file?

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/index.html
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/index.html:259]: Good attempt on your look and feel.

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js]: Is this absolutely necessary?

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js]: what is iDex?

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js:60]: Some of your test for your readFile seem like validation test. Why is this so?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js:23]: Is your readFile actually reading the **_file_**?

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js]: Why are you doing a beforeEach() here?

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js]: Please use template literals here instead of concatenating this string.

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js]: Please use template literals here instead of concatenating this string.

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js]: Please use template literals here instead of concatenating this string or put the entire object in a file and reference it.

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js]: apropriate or appropriate?  - Please correct this typo here and also where it reoccurs

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js]: you are missing _return_ after _should_

**Review by andela-moe — COMMENTED** [review]
Please review comments

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (thebere) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: Not absolutely, but necessary for Circle CI to work.
Circle CI, seem not to be in harmony with ES6 unless run in strict mode.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (thebere) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js:23]: The reason I've used this approach is to write a method that works in any environment.
The idea is to read a file. Because, I only expect a string (json file) that can be read to a JavaScript object. So, I just parse it as json, if it can read that, it returns the content, else it return false.
As such, this module can be used on node applications both server and client side.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (thebere) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: Noted. Will do that

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (thebere) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: Okay. Will work on that

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (thebere) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: Will fix that

File: 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/outcome/README.md
Comment (thebere) [reply 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/outcome/README.md:8]: Thanks

File: 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md
Comment (thebere) [reply 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md:17]: Yes. I got two fellows to review what I have done on a PR before the merge

File: 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md
Comment (thebere) [reply 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md:8]: Thanks

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md
Comment (thebere) [reply 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md:9]: Thanks

File: 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md
Comment (thebere) [reply 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md:7]: Thanks

File: 06-building-a-program-start-to-finish/using-design-patterns/output/README.md
Comment (thebere) [reply 06-building-a-program-start-to-finish/using-design-patterns/output/README.md:75]: Thanks

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (thebere) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: Noted. Fixing that right away

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (thebere) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: Yes, I did. However, these methods, are actually not meant to return values.
If replaced with the function keyword the error disappears though.
e.g:
[].map(() => {
// throws error
});

[].map(function () {
// is fine
})

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/index.html
Comment (thebere) [reply 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/index.html:259]: Thanks

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (thebere) [reply 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js]: iDex is, what I named this application. What do you think?

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (thebere) [reply 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js]: I need to create indexes for those files/books, to run some of the test cases.

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (thebere) [reply 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js:60]: Because, I wrote the readfile to just parse a json string to a JS object if it is a valid json or return false if it is unable to do so.

I the condition makes it seem like a validation, which may still be the case.

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (thebere) [reply 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js]: Okay.

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (thebere) [reply 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js]: Okay

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (thebere) [reply 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js]: Thank you

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (thebere) [reply 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js]: Thanks

File: 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md
Comment (andela-moe) [reply 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md:17]: Yes I see that, but what D(_level_) fellow were you to get reviews from?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: I see you are returning _this_. What does _**this**_ mean in that scenario?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: I see you are returning _this_. What does _**this**_ mean in that scenario?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: I see you are returning _this_. What does _**this**_ mean in that scenario?

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js:62]: Generally the language used on your tests should be understood by both the developer and the customer. It helps you shape the requirements and testing needed, at the level of the customer's understanding.

```
describe("Calculator: add", () => {

    it("should be able to add 2 numbers together", () => {
        var total = add(1, 2);
        expect(total).toBe(3);
    });

    it("should be able to add 3 numbers together", () => {
        var total = add(1, 2, 3);
        expect(total).toBe(6);
    });
});
```
The result of this will read:
**_Calculator: add_**
    _should be able to add 2 numbers together_
   _should be able to add 3 numbers together_

Please review the language used for your validateFile test. Paying more attention to the language used on your "it" and 'describe' functions.

**Review by andela-moe — COMMENTED** [review]
Please review comment

File: 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md
Comment (thebere) [reply 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md:17]: Okay. I see. It wasn't specified. These D0b fellows

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js
Comment (thebere) [reply 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js:62]: Okay. Thanks

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (thebere) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: From study, by default functional methods such as `map` return an implicit `this`. Which is used to support method chaining.
In this scenario, I am returning this `tokenIndex[eachWord]`, thereby making it possible to chain other methods on the map method. I think what this means is, whether I explicitly do it or not, it is done internally. But I have done it to avoid Eslint flags.

I have however, changed it to explicitly return `tokenIndex[eachWord]`

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js
Comment (thebere) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: From study, by default functional methods such as `map` return an implicit `this`. Which is used to support method chaining.
In this scenario, I am returning this `tokenIndex[eachWord]`, thereby making it possible to chain other methods on the map method. I think what this means is, whether I explicitly do it or not, it is done internally. But I have done it to avoid Eslint flags.

I have however, changed it to explicitly return `tokenIndex[eachWord]`

**Review by andela-moe — APPROVED** [review]
Good work

**Pull request merged**
